readded everything (#104)
MrLongbottom/P10-Article: sections/appendix/metrics.tex

\subsection{Topic coherence}\label{app:topic_coherence}
The equations, explanations, and values of the hyperparameters in this section are based on~\citet{Syed2017coherence} and the \textit{gensim} python package\footnote{\url{https://radimrehurek.com/gensim/}}.
Calculating topic coherence requires the following steps:

\begin{enumerate}
\item Topic-word segmentation into word set pairs
\item Word and word pair probability calculation
\item Word set confirmation measure
\item Aggregation of confirmation measures
\end{enumerate}

For segmentation, a set of word pairs $S$ is created, which pairs each word in the top-N most probable words $W$ in a specific topic $t$ with all other words in $W$.
$S$ is defined by \autoref{eq:set_of_word_pairs}.

\begin{equation}\label{eq:set_of_word_pairs}
	S = \{(W', W*)|W' = \{w_i\};w_i \in W;W* = W\}
\end{equation}

> Before calculating word probabilities, a sliding window of size $s$ where $s =110$ is used to create a set of subdocuments $D_s$ over the document set $D$.

Reviewer: ```suggestion
Before calculating word probabilities, a sliding window of size $s$ where $s =110$ is used to create a set of subdocuments $D_s$ over the document set $D$.
We choose not to change the sliding window size $s$ to be able to compare against other papers, but this is a hyperparameter, which could provide better results for our dataset if changed. 
```

> These subdocuments are used rather than the normal documents to capture some degree of word proximity.

Author: ```suggestion
\todo{talk about how sliding window works, and how it influences this metric}
These subdocuments are used rather than the normal documents to capture some degree of word proximity.
```

Word probabilities are calculated based on how many documents, within the document set $D_s$, they occur in.
$p(w_i)$ is the number of subdocuments in which the word $w_i$ occurs divided by $|D_s|$.
$p(w_i, w_j)$ is the number of subdocuments in which both words occur divided by $|D_s|$. 

> We create a \gls{npmi} matrix of size $|W|\times|W|$, with one entry per word pair combination in $W$.

Reviewer 2: ```suggestion
As part of the word set confirmation measure we create a \gls{npmi} matrix of size $|W|\times|W|$, with one entry per word pair combination in $W$.
```
Prøver at lede ind til at vi nu bygger op til at have en confirmation measure

\begin{equation}\label{eq:coherence_2}
	\text{NPMI}(w_i,w_j) =  \frac{\log\frac{p(w_i,w_j) + \epsilon}{p(w_i)\cdot p(w_j)}}{-log(p(w_i,w_j) + \epsilon)}
\end{equation}
\noindent where $\epsilon$ is a low number ($10^{-12}$) used to avoid $log(0)$.

> The \gls{npmi} matrix describes how much each word occurs alone versus together with other words.

Reviewer: ```suggestion
The \gls{npmi} matrix describes how much each word in the topic co-occurs with the other words.
```
